test/replayablestream.js

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2018 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it */
+'use strict';
+
+const assert = require('assert');
+const fs = require('fs');
+const stream = require('stream');
+
+const ReplayableStream = require('../lib/replayablestream.js');
```

Comment:
It would be nice if Node.js provided a test for compliance to the `Stream` `Readable` and `Writeable` prototypes that we could use to test this and the PassThroughDuplex. It would help alert us to API changes on them as well. Alas I couldn't find one.